pkg/stanza/operator/parser/syslog/syslog.go

```diff
@@ -93,28 +111,52 @@ func (c Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
 	}
 
 	return &Parser{
-		ParserOperator: parserOperator,
-		protocol:       c.Protocol,
-		location:       location,
+		ParserOperator:               parserOperator,
+		protocol:                     c.Protocol,
+		location:                     location,
+		enableOctetCounting:          c.EnableOctetCounting,
+		nonTransparentFramingTrailer: c.NonTransparentFramingTrailer,
 	}, nil
 }
 
-func buildMachine(protocol string, location *time.Location) (sl.Machine, error) {
-	switch protocol {
+// parseFunc a parseFunc determines how the raw input is to be parsed into a syslog message
+type parseFunc func(input []byte) (sl.Message, error)
+
+func (s *Parser) buildParseFunc() (parseFunc, error) {
+	switch s.protocol {
 	case RFC3164:
-		return rfc3164.NewMachine(rfc3164.WithLocaleTimezone(location)), nil
+		return func(input []byte) (sl.Message, error) {
+			machine := rfc3164.NewMachine(rfc3164.WithLocaleTimezone(s.location))
+			return machine.Parse(input)
+		}, nil
 	case RFC5424:
-		return rfc5424.NewMachine(), nil
+		switch {
+		// Octet Counting Parsing RFC6587
+		case s.enableOctetCounting:
+			return newOctetCountingParseFunc(), nil
+		// Non-Transparent-Framing Parsing RFC6587
+		case s.nonTransparentFramingTrailer != nil:
```

Comment:
I'm not super familiar with the protocol, but I thought I remember reading that it's possible to use both of these. Is this switch statement forcing the assumption that they are mutually exclusive?

Reply:
I made them mutually exclusive due to how the library treats it. You can either have an octet counting parser or a non-transparent-framing parser. 

I'm reading through the RFC and [this](https://www.rfc-editor.org/rfc/rfc6587#section-3.4) section makes it sound like non-transparent-framing is an older way of doing things and octet counting is newer. I'm not sure if that means they are two ways to accomplish the same thing or if they can be used together.

Reply:
Alright, make sense.